sdk/core/azure-core/src/main/java/com/azure/core/exception/AzureException.java

```diff
@@ -25,9 +25,22 @@ public AzureException(final String message) {
      * Initializes a new instance of the AzureException class.
      *
      * @param message The exception message.
-     * @param cause The Throwable which caused the creation of this AzureException.
+     * @param cause The {@link Throwable} which caused the creation of this AzureException.
      */
     public AzureException(final String message, final Throwable cause) {
         super(message, cause);
     }
+
+    /**
+     * Initializes a new instance of the AzureException class.
+     *
+     * @param message The exception message.
+     * @param cause The {@link Throwable} which caused the creation of this AzureException.
+     * @param enableSuppression Whether suppression is enabled or disabled.
+     * @param writableStackTrace Whether the exception stack trace will be filled in.
+     */
+    public AzureException(final String message, final Throwable cause, final boolean enableSuppression,
```

Comment:
Add the other 2 overloads too.
`public AzureException()`
`public AzureException(Throwable cause)`.

Reply:
Should these overloads also be added into the other based exception types such as `HttpResponseException` and `HttpRequestException`?

Reply:
It's better to provide all constructor overloads and let the caller decide which ctor they want to use. Since all we are doing is delegating to the super ctor, I think it's better to include them in all exception types we create.

Reply:
Added additional constructors to `AzureException`, `HttpRequestException`, and `HttpResponseException` as they were the only exception types in Azure Core which that have usage.